sparse/unit_test/Test_Sparse_sptrsv.hpp

```diff
@@ -38,1111 +38,557 @@ using namespace KokkosKernels;
+
+        sptrsv_symbolic(&kh, row_map, entries);
+        Kokkos::fence();
+
+        sptrsv_solve(&kh, row_map, entries, values, rhs, lhs);
+        Kokkos::fence();
+
+        scalar_t sum = 0.0;
+        Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)),
+                                ReductionCheck(lhs), sum);
+        EXPECT_EQ(sum, lhs.extent(0));
+
+        kh.destroy_sptrsv_handle();
       }
-      EXPECT_TRUE(sum == scalar_t(X.extent(0)));
 
-      khLd.destroy_sptrsv_handle();
-      khUd.destroy_sptrsv_handle();
-    }
+#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+      if (std::is_same<size_type, int>::value &&
+          std::is_same<lno_t, int>::value &&
+          std::is_same<typename device::execution_space, Kokkos::Cuda>::value) {
+        Kokkos::deep_copy(lhs, ZERO);
+        KernelHandle kh;
+        bool is_lower_tri = false;
+        kh.create_sptrsv_handle(SPTRSVAlgorithm::SPTRSV_CUSPARSE, nrows,
+                                is_lower_tri);
+
+        sptrsv_symbolic(&kh, row_map, entries, values);
+        Kokkos::fence();
+
+        sptrsv_solve(&kh, row_map, entries, values, rhs, lhs);
+        Kokkos::fence();
+
+        scalar_t sum = 0.0;
+        Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)),
+                                ReductionCheck(lhs), sum);
+        EXPECT_EQ(sum, lhs.extent(0));
+
+        kh.destroy_sptrsv_handle();
+      }
 #endif
-  }
-}
 
-template <typename scalar_t, typename lno_t, typename size_type,
-          typename device>
-void run_test_sptrsv_streams(int test_algo, int nstreams) {
-  using RowMapType             = Kokkos::View<size_type *, device>;
-  using EntriesType            = Kokkos::View<lno_t *, device>;
-  using ValuesType             = Kokkos::View<scalar_t *, device>;
-  using RowMapType_hostmirror  = typename RowMapType::HostMirror;
-  using EntriesType_hostmirror = typename EntriesType::HostMirror;
-  using ValuesType_hostmirror  = typename ValuesType::HostMirror;
-  using execution_space        = typename device::execution_space;
-  using memory_space           = typename device::memory_space;
-  using KernelHandle = KokkosKernels::Experimental::KokkosKernelsHandle<
-      size_type, lno_t, scalar_t, execution_space, memory_space, memory_space>;
-  using crsMat_t = CrsMatrix<scalar_t, lno_t, device, void, size_type>;
+#if defined(KOKKOSKERNELS_ENABLE_SUPERNODAL_SPTRSV)
+      const scalar_t FIVE    = scalar_t(5);
+      const size_type nnz_sp = 14;
+      {
+        // U in csr
+        auto ut_fixture = get_5x5_ut_fixture();
+        row_map_view_t hUrowptr;
+        cols_view_t hUcolind;
+        values_view_t hUvalues;
+
+        // first row -> first supernode
+        // second row -> first supernode
+        // third row -> second supernode
+        // fourth row -> third supernode
+        // fifth row -> fourth supernode
+
+        compress_matrix(hUrowptr, hUcolind, hUvalues, ut_fixture);
```

Comment:
I am guessing this is compressing the dense matrix?

Reply:
Yep! See my comment above.